### DISCUSSION TITLE
ask; navigation

### DISCUSSION THREAD

**Discussion by moonwalkin** [android/nowinandroid#1581] [Q&A] [answered]
Each of the feature modules has navigation, but I don't see the navigation library connection anywhere except for the app module. I looked at all the plugins, but I didn't see it there either. Please help me to understand how it works.

**Comment by TerryQuiet** [comment]
each feature module applies this convention plugin
https://github.com/android/nowinandroid/blob/main/build-logic/convention/src/main/kotlin/AndroidFeatureConventionPlugin.kt

**Reply by moonwalkin** [reply]
I saw this, but where is the navigation connected? i saw only hilt navigation

**Reply by TerryQuiet** [reply]
not sure, but maybe you're looking for this
https://github.com/android/nowinandroid/blob/main/feature/interests/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/interests/navigation/InterestsNavigation.kt
every feature module have extension function on **NavGraphBuilder** and **NavController**
and NavGraph uses those extension functions to build the app graph.
https://github.com/android/nowinandroid/blob/main/app/src/main/kotlin/com/google/samples/apps/nowinandroid/navigation/NiaNavHost.kt

**Reply by moonwalkin** [reply]
I understand that each module has extensions but I'm interested in something else

To make an extension on some class, there must be a dependency, in this case: androidx.navigation:navigation-compose.

Modules do not have this dependency.
There is a plugin AndroidFeatureConventionPlugin that used to have this dependency
https://github.com/android/nowinandroid/blob/8fc2e15e73de9de90ef535ffb2ca62c051d865a8/build-logic/convention/src/main/kotlin/AndroidFeatureConventionPlugin.kt (old version of the file)
Now this dependency is gone, but it still works
https://github.com/android/nowinandroid/blob/main/build-logic/convention/src/main/kotlin/AndroidFeatureConventionPlugin.kt (current file version)

I hope you understand what I mean

**Reply by TerryQuiet** [reply, marked as answer]
https://mvnrepository.com/artifact/androidx.hilt/hilt-navigation-compose/1.2.0 
we can see here that hilt comes with those dependencies.
also we can use gradle task too see all the transitive dependencies `./gradlew :feature:foryou:dependencies `

**Comment by Jaehwa-Noh** [comment]
Hilt-navigation-compose has the dependency of navigation-compose, so that it is fine that don't use navigation-compose.
